src/main/java/fi/aalto/cs/apluscourses/ui/ideactivities/OverlayPane.java

```diff
@@ -93,6 +94,9 @@ protected void paintComponent(Graphics graphics) {
     Graphics2D g = (Graphics2D) graphics.create();
     g.setComposite(AlphaComposite.SrcOver.derive(0.7f));
     g.setColor(Color.BLACK); // using JBColor.BLACK is wrong here
+    g.setRenderingHint(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON);
+    g.setRenderingHint(RenderingHints.KEY_RENDERING, RenderingHints.VALUE_RENDER_QUALITY);
+    g.setRenderingHint(RenderingHints.KEY_INTERPOLATION, RenderingHints.VALUE_INTERPOLATION_BILINEAR);
```

Comment:
Is it known if these may hit the performance on less efficient machines?

Reply:
IntelliJ uses these too so maybe it's not a problem

Reply:
Less efficient machines will not run IntelliJ. :>
In all seriousness, it won't affect performance in any noticeable way.